src/utils/include/utils/device_qualifier.hpp

```diff
@@ -1,7 +1,11 @@
 #ifndef UTILS_DEVICE_QUALIFIER_HPP
 #define UTILS_DEVICE_QUALIFIER_HPP
 
-#if defined(__CUDACC__)
+#if defined(__CUDACC__) || defined(__HIPCC__)
+#define GPUCOMPILER
```

Comment:
I think it would be more consistent to do

```c++
#if defined(__CUDACC__)
#define DEVICE_COMPILER cudacc
#elif  defined(__HIPCC__)
#define DEVICE_COMPILER hipcc
#endif 

#if defined(DEVICE_COMPILER)
#define DEVICE_THROW(E) throw (E)
#else 
#define DEVICE_THROW(E)
#endif 
```